lib/libc/Kconfig

```
@@ -35,8 +35,8 @@ config NEWLIB_LIBC_SUPPORTED
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr") || (NATIVE_LIBRARY)
-	depends on !(CPP && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr")
+	depends on ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || TOOLCHAIN_SUPPORTS_PICOLIBC_MODULE || NATIVE_LIBRARY
+	depends on !(CPP && (("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || TOOLCHAIN_SUPPORTS_PICOLIBC_MODULE))
```

upon second review I was looking at https://github.com/zephyrproject-rtos/sdk-ng/blob/main/cmake/zephyr/Kconfig and discovered that the reason for this code is to handle Zephyr SDK 0.15.x.

Thus we should not change this code.
If a toolchain supports picolibc (built-in or as module) then it should have the following code in its toolchain Kconfig.
```
config PICOLIBC_SUPPORTED
	default y
```
```suggestion
	depends on ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr") || (NATIVE_LIBRARY)
	depends on !(CPP && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr")
```